### DISCUSSION TITLE
Future of Prism with Sponsor Connect

### DISCUSSION THREAD

**Discussion by jaredballen** [PrismLibrary/Prism#2223] [General]
I am wondering what the future of Prism is. The latest release notes indicate that all packages moving forward will be moving to Sponsor Connect. Does this mean that there will be no more artifacts published to nuget.ord (after 8.0.0.1909)? This release also removed support for `IAutoInitialize` and `AutoRegisterForNavigation`. These have been moved into the closed source for Prism.Magician. Can we expect other features in the Prim mainline to be moved into closed source as well?

**Comment by brianlagunas** [comment]
Prism itself will remain on NuGet freely available to everyone. The only packages moving to Sponsors are the CI builds and custom tooling. There will be certain features or tooling that will only be made available to Sponsors of Dan and myself.  The IAutoInitialize features were removed due to performance reasons as they relied heavily on reflection. Prism.Magician does the same thing but much better and does not rely on reflection to do it.  Not everything we do will be OSS and to access those tools and/or features you'll need to be a sponsor.  There are no current plans for any other feature to be removed from Prism and moved into a Sponsor requirement. Anything new may be a Sponsor only feature.

**Comment by dansiegel** [comment]
@jaredballen all official releases will continue being published to nuget.org and all of the assemblies and packages along with the release notes will continue to be on GitHub. We believe it's not too much to ask that if you need a critical bug fix that's been merged but is not part of an official release that you step up to help keep the project sustainable by becoming a sponsor.

Special tooling like what I did for VS 4 Mac had previously been freely available. In the past I may have released the Prism.Magician for free as well, and to be honest I would love to make it freely available for everyone. However in order to do that I need the community to step up in a big way to help me fund the thousands of hours of development that go into that special productivity tooling. Until then that tooling will remain available only to people who are stepping up as sponsors.

As @brianlagunas was mentioning on the IAutoInitialize & AutoRegisterForNavigation, both of these features had a Reflection cost associated with them that affected every app while the Magician is able to do this with a minimal build time impact.
